src/pip/_internal/commands/install.py

```diff
@@ -323,6 +324,15 @@ def run(self, options, args):
                 reqs, check_supported_wheels=not options.target_dir
             )
 
+            # Download any requirements which were only fetched by metadata.
+            # Let's download to a temporary directory.
+            tmpdir = TempDirectory(kind="unpack", globally_managed=True).path
+            download_completer = PartialRequirementDownloadCompleter(
+                session,
+                progress_bar=options.progress_bar,
+                download_dir=tmpdir)
+            download_completer.complete_requirement_downloads(requirement_set)
```

Comment:
Given that it's a class w/ a single method that's called immediately after initialization, it's probably better to make this into a function.

It helps make it clear that it's a single logical unit.

Reply:
Absolutely!